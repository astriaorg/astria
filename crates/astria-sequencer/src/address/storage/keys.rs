pub(in crate::address) const BASE_PREFIX_KEY: &str = "address/prefixes/base";
pub(in crate::address) const IBC_COMPAT_PREFIX_KEY: &str = "address/prefixes/ibc_compat";

#[cfg(test)]
mod tests {
    use super::*;

    const COMPONENT_PREFIX: &str = "address/";

    #[test]
    fn keys_should_not_change() {
        insta::assert_snapshot!(BASE_PREFIX_KEY);
        insta::assert_snapshot!(IBC_COMPAT_PREFIX_KEY);
    }

    #[test]
    fn keys_should_have_component_prefix() {
        assert!(BASE_PREFIX_KEY.starts_with(COMPONENT_PREFIX));
        assert!(IBC_COMPAT_PREFIX_KEY.starts_with(COMPONENT_PREFIX));
    }
}
