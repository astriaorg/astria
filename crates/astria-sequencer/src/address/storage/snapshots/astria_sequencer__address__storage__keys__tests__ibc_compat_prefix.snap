---
source: crates/astria-sequencer/src/address/storage/keys.rs
expression: IBC_COMPAT_PREFIX
---
address/prefixes/ibc_compat
